pkg/handler/collector/deps_dev/deps_dev_test.go

```diff
@@ -462,3 +470,48 @@ func TestProjectKey(t *testing.T) {
 		})
 	}
 }
+
+func TestDepsCollector_collectAdditionalMetadata(t *testing.T) {
+	tests := []struct {
+		testName     string
+		pkgType      string
+		namespace    *string
+		name         string
+		version      *string
+		pkgComponent *PackageComponent
+		wantErr      bool
+	}{
+		{
+			testName:     "npm package with .git suffix",
+			pkgType:      "npm",
+			namespace:    ptrfrom.String("@webassemblyjs"),
+			name:         "wasm-parser",
+			version:      ptrfrom.String("1.11.6"),
+			pkgComponent: &PackageComponent{},
+			wantErr:      false,
+		},
+		{
+			testName:     "golang package without .git suffix",
+			pkgType:      "golang",
+			namespace:    ptrfrom.String("github.com/google"),
+			name:         "wire",
+			version:      ptrfrom.String("v0.5.0"),
+			pkgComponent: &PackageComponent{},
+			wantErr:      false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctx := context.Background()
+			c, err := NewDepsCollector(ctx, toPurlSource([]string{}), false, true, 5*time.Second)
+			if err != nil {
+				t.Errorf("NewDepsCollector() error = %v", err)
+				return
+			}
+			if err := c.collectAdditionalMetadata(context.Background(), tt.pkgType, tt.namespace, tt.name, tt.version, tt.pkgComponent); (err != nil) != tt.wantErr {
```

Comment:
You are right, `collectAdditionalMetadata` has logs instead of errors, so the tests are now checking the logs to see whether they contain the expected error. Again, this is in PR #2009.